### DISCUSSION TITLE
Instantiate elements after DOM update

### DISCUSSION THREAD

**Discussion by mvaliolahi** [alpinejs/alpine#2373] [1. Help] [answered]
Is there a solution for things like DOMContentLoaded?

Problem: after the video list update element needs to be instantiated again.

            document.addEventListener('DOMContentLoaded', () => {
                const players = Array
                    .from(document.querySelectorAll('.video-player'))
                    .map(p => new Plyr(p, {
                        active: false,
                        language: 'auto',
                        update: false
                    }));
            });

**Comment by SimoTod** [comment, marked as answer]
You need to post a simplified but full example. In general, elements created dynamically after page load are automatically instantiated by alpine but I assume something is not working well for you.

**Reply by mvaliolahi** [reply]
I use 'plyrJs' for video elements.
After page load everything is ok and video tags populate using the plyrJs, but when loadMoreData calls and the posts come from API  for new items the plyrJs are not called yet and tags are normal html5 style.

It is not an AlpineJs issue, in fact, I need a clean solution for the call  ` new Plyr()` after new elements are created using AlpineJs, my solution is not efficient :)

```html
<div class="w-full" x-data="PostListLogic">
    <template x-for="(post, index) in posts" :key="index">
        <div>
		<video :src="post.url" controls class="w-full video-player"></video>
        </div>
    </template>

    <div x-intersect="loadMoreData()">
    </div>

</div>


<script src="https://cdn.plyr.io/3.6.9/plyr.js"></script>
<script>
    document.addEventListener('DOMContentLoaded', () => {
        const players = Array
                    .from(document.querySelectorAll('.video-player'))
                    .map(p => new Plyr(p, {
                        active: false,
                        language: 'auto',
                        update: false
	}));
    });

	function PostListLogic() {
		return {
			posts: [{}, {}],
                       loadMoreData: function() {
                              const response = api call ...
                              this.posts   = this.posts.concat(response.data.data);
                       },
		};
	}
</script>
```

**Reply by SimoTod** [reply]
You can put an x-init on each video tag generated by x-for and instantiate the player for that tag
Something like `x-init="new Plyr($el, {...})"`

Just make sure the order is correct. Plyr must be included before Alpine (you can defer it as well, up to you).

**Reply by mvaliolahi** [reply]
Thanks @SimoTod
